### ISSUE TITLE
Training Split results in poorer models

### ISSUE THREAD

**Issue by bart-vanvlerken** [facebookexperimental/Robyn#755]
Hi everyone! Hope you're all doing well. I'd like your feedback on the following issue: it is a good practice to apply a train/val/test split to see how the model performs on out-of-sample data, however the models resulting from `ts_validation = TRUE` are significantly worse (in my case). 

I have about 15 months worths of daily data, and since there is a massive peak at the end of every year, I'd like to see how the model performs on the last three months worth of data (so it can use the first year of data for training). I've therefore set `train_size = c(0.85, 0.9)`. The resulting models all have a very low R2, similar to the one in the figure below.

![model](https://github.com/facebookexperimental/Robyn/assets/118725564/eaaf4c1e-65df-4bd0-969f-948efa00ecda)

Because the model optimizes for the validation set, a lot of the variance in the training data is not being captured and is instead attributed to the intercept. The marketing channels suffer from this as well and as a result, the CPAs are way higher than anticipated which is very difficult to communicate to stakeholders. With `ts_validation = FALSE`, the models look a lot better, but I cannot be sure how good they actually are since they have not been tested on out-of-sample data. An example model can be seen below. I'd appreciate anyone's two cents on this matter: should I change anything or leave it the way it is?

![model_train](https://github.com/facebookexperimental/Robyn/assets/118725564/b2689756-fecd-459c-986c-fa7def6e12bd)

**Comment by gufengzhou** [comment]
This is a very interesting case, thanks for sharing. With this dataset, when setting training size to .85-.9, the peak is actually not informative for the validation set prediction, so the optimizer does't make the "effort" to fit that peak. In your case, it's indeed better to not use ts validation. If you have 2+ years data and include one peak in the validation set, I can imagine it works better then.
